### ISSUE TITLE
Ubuntu 18.4 LTS service start problem

### ISSUE THREAD

**Issue by brevius** [checktheroads/hyperglass#36]
Hi, 
It looks there's a problem with file path once started as a service:
Apr 17 09:11:45 hyperglass hyperglass[5973]:   File "/usr/local/lib/python3.6/dist-packages/hyperglass/configuration/models/web.py", line 236, in Web
Apr 17 09:11:45 hyperglass hyperglass[5973]:     logo: Logo = Logo()
Apr 17 09:11:45 hyperglass hyperglass[5973]:   File "pydantic/main.py", line 283, in pydantic.main.BaseModel.__init__
Apr 17 09:11:45 hyperglass hyperglass[5973]: pydantic.error_wrappers.ValidationError: 2 validation errors for Logo
Apr 17 09:11:45 hyperglass hyperglass[5973]: light
Apr 17 09:11:45 hyperglass hyperglass[5973]:   '**None**/static/images/hyperglass-light.png' does not exist (type=value_error)
Apr 17 09:11:45 hyperglass hyperglass[5973]: dark
Apr 17 09:11:45 hyperglass hyperglass[5973]:   '**None**/static/images/hyperglass-dark.png' does not exist (type=value_error)
Apr 17 09:11:45 hyperglass systemd[1]: hyperglass.service: Main process exited, code=exited, status=1/FAILURE
Apr 17 09:11:45 hyperglass systemd[1]: hyperglass.service: Failed with result 'exit-code'.

t.

**Comment by akaczm** [comment]
I have noticed this code that seems to determine the filepaths
```
config_path = Path(os.environ["hyperglass_directory"])
    base_path = [v for v in value.split("/") if v != ""]

    if base_path[0] not in ("images", "custom"):
        raise ValueError(
            f"Logo files must be in the 'custom/' directory of your hyperglass directory. Got: {value}"
        )

    if base_path[0] == "custom":
        file = config_path / "static" / "custom" / "/".join(base_path[1:])

    else:
        file = config_path / "static" / "images" / "/".join(base_path[1:])

    log.error(file)
```
It refers to a hyperglass_directory environment variable? 
Trying to define within the systemd unit file still returns None in the path

**Comment by checktheroads** [comment]
Hi there,

This can happen if the `/etc/hyperglass` directory (which requires root privileges) is chosen during setup, but running the `hyperglass` command without `sudo`. It can also happen if hyperglass is installed without root privileges (`pip3 install hyperglass` vs `sudo pip3 install hyperglass`, and the `/etc/hyperglass’ path is chosen.

Essentially what’s happening is hyperglass tries to determine which directory is used on startup. If it can access `/etc/hyperglass` and `~/hyperglass` doesn’t exist, `None` ends up being set, which doesn’t exist. I’ve been working on handling this scenario and believe I have a solution, but I haven’t pushed it into a release yet (still testing).

In the meantime, it sounds to me like scenario 2 from the above is what’s affecting you, since you’re not running the `hyperglass` command directly. If that’s the case, it _should_ be as easy as `pip3 uninstall hyperglass && sudo pip3 install hyperglass`

**Comment by brevius** [comment]
Hi,
Thanks for checking. In fact I installed hyperglass using sudo on Openstack system (so in /home/ubuntu directory). Starting from CLI is fine (with sudo). I'll try to reinstall all the stuff once again.

**Comment by brevius** [comment]
Hi,
I reinstalled hyperglass as you suggested (pip3 uninstall hyperglass && sudo pip3 install hyperglass), next (all with sudo) hyperglass setup with /etc/hyperglass, systemctl enable and build-ui
Now service started without problems.
Thanks!

**Comment by brevius** [comment]
It looks it is bad day for me. I encoutered another problem (empty `hyperglass.yaml` file):
```
[2020-04-20 20:06:59 +0000] [5590] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.6/site-packages/uvicorn/protocols/http/httptools_impl.py", line 385, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/fastapi/applications.py", line 142, in __call__
    await super().__call__(scope, receive, send)  # pragma: no cover
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/applications.py", line 134, in __call__
    await self.error_middleware(scope, receive, send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/middleware/errors.py", line 178, in __call__
    raise exc from None
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/middleware/errors.py", line 156, in __call__
    await self.app(scope, receive, _send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/middleware/cors.py", line 140, in simple_response
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/exceptions.py", line 73, in __call__
    raise exc from None
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/exceptions.py", line 62, in __call__
    await self.app(scope, receive, sender)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/routing.py", line 590, in __call__
    await route(scope, receive, send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/routing.py", line 208, in __call__
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/fastapi/routing.py", line 133, in app
    raw_response = await dependant.call(**values)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hyperglass/api/routes.py", line 35, in query
    async with Webhook(params.logging.http.provider) as hook:
AttributeError: 'NoneType' object has no attribute 'provider'
```

**Comment by checktheroads** [comment]
Sorry about that - major bug I missed. This should be fixed in 1.0.0-beta29
